### PR TITLE
Update LoginHookListener.php

### PR DESCRIPTION
**Pull request by TomSpidge**
IPv6 update - parses both incoming and whitelisted addresses in IPv4 or IPv6 format.

### CODE REVIEW

File: lib/LoginHookListener.php [hunk 1]
```diff
@@ -45,29 +45,62 @@ public function __construct(IConfig $config,
...
+                        $binMask .= "e";
+                        break;
+                }
+                $binMask = str_pad($binMask, 32, '0');
+                $binMask = pack("H*", $binMask);
 
-		if ($bits === '') {
-			$bits = 32;
-		}
+                return ($ip & $binMask) == $subnet;
```
Comment (mario) [inline lib/LoginHookListener.php]: Missing another "=" maybe?

File: lib/LoginHookListener.php [hunk 2]
```diff
@@ -45,29 +45,62 @@ public function __construct(IConfig $config,
...
+            }
+            return false;
```
Comment (mario) [inline lib/LoginHookListener.php]: New line before this?

File: lib/LoginHookListener.php (hunk 1 above)
Comment (mario) [inline lib/LoginHookListener.php]: Extra empty line there? Maybe remove?

File: lib/LoginHookListener.php [hunk 3]
```diff
@@ -45,29 +45,62 @@ public function __construct(IConfig $config,
...
+         /**
 	 * @param $ip
 	 * @param $range
 	 * @return bool
-	 * @copyright https://stackoverflow.com/questions/594112/matching-an-ip-to-a-cidr-mask-in-php-5/594134#594134
+	 * @copyright (IPv4) https://stackoverflow.com/questions/594112/matching-an-ip-to-a-cidr-mask-in-php-5/594134#594134
+	 * @copyright (IPv6) MW. https://stackoverflow.com/questions/7951061/matching-ipv6-address-to-a-cidr-subnet via https://www.phpclasses.org/browse/file/70429.html
 	 */
-	private function matchCidr($ip, $range) {
-		if ($range === '') {
```
Comment (rullzer) [inline lib/LoginHookListener.php]: why is this gone?

Comment (TomSpidge) [reply lib/LoginHookListener.php]: Not sure, it's been over a year. Probably should still be there.

**Comment by Altahrim** [comment]
Changes already merged

**Pull request closed without merging**